src/main/java/core/shop/service/impl/CalculateQuantityServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.shop.service.impl;
+
+import core.shop.model.ActivityType;
+import core.shop.model.FruitRecord;
+import core.shop.service.CalculateQuantityService;
+import java.util.List;
+
+public class CalculateQuantityServiceImpl implements CalculateQuantityService {
+    @Override
+    public int calculateQuantity(List<FruitRecord> fruits, String fruitName) {
```

Comment:
It is necessary to divide this logic into several classes. Create your own handler for each operation, for example BalanceHandler. It will have one handle(Operation) method and will do the necessary actions.

Reply:
It will also be necessary to create an OperationStrategy class, which, depending on the type of operation, will return the necessary handler for us to process

Reply:
It is worth creating a Storage class, where the Map will be stored, to which fruits will first be added in accordance with the operations. And then a report is generated from it.